Tests/FluentAssertions.Specs/Collections/GenericCollectionAssertionsSpecs.cs

```diff
@@ -318,14 +318,15 @@ public void When_asserting_an_empty_collection_contains_a_single_item_matching_a
         {
             // Arrange
             IEnumerable<int> collection = Enumerable.Empty<int>();
-            Expression<Func<int, bool>> expression = item => item == 2;
+            var targetValue = 2;
```

Comment:
Why extracting `2` to a variable?

Reply:
Basically, to assert that the lambda output formatting is invoked properly. Previously, having this syntax would have rendered the message "Expected collection to contain a single item matching (item == value(FluentAssertions.Specs.Collections.GenericCollectionAssertionsSpecs+<>c__DisplayClass21_0).targetValue)"

Reply:
It seems that `When_first_level_properties_are_tested_for_equality_against_constant_expressions_then_output_should_contain_values_of_constant_expressions` is responsible for asserting that captured constants in expressions are properly reduced, and not a concern these tests need to exercise.

But I do like that you're replacing the dynamic part of `expectedMessage` 👍

Reply:
`When_first_level_properties_are_tested_for_equality_against_constant_expressions_then_output_should_contain_values_of_constant_expressions` does indeed test the formatting, but the previous use of `.Body` resulted in `PredicateLambdaExpressionValueFormatter` not being used at all. I guess, in a sense, it turns the test into more of an integration test, but that's the case for all tests asserting exception messages.

The dynamic part + the use of an expression with a constant `(item => item == 2)` is what hid the problem from view. That was my reasoning behind this change. If you which me to reset it to `item => item == 2` I will gladly do so.

Reply:
If you change it to
```cs
Expression<Func<int, bool>> expression = item => item == 2;
```
and
```cs
"Expected collection to contain a single item matching (item => item == 2), but the collection is empty."
```

that's perfectly fine with me 👍 

You could add another test that asserts on how the message is formatted, but I don't have a strong opinion about that.